resources/asciidoctor/spec/elastic_compat_preprocessor_spec.rb

```diff
@@ -364,4 +370,61 @@
     DOCBOOK
     expect(actual).to eq(expected.strip)
   end
+
+  def stub_file_opts
+    return {
+      'copy_snippet' => proc { |uri, source| },
+      'write_snippet' => proc { |uri, source| },
+    }
+  end
+
+  [
+      %w[CONSOLE console],
+      %w[AUTOSENSE sense],
+      %w[KIBANA kibana],
+  ].each do |name, lang|
+    it "transforms #{name} comments into a listing with the #{lang} language" do
+      input = <<~ASCIIDOC
+        == Example
+        [source,js]
+        ----
+        foo
+        ----
+        // #{name}
+      ASCIIDOC
+      expected = <<~DOCBOOK
+        <chapter id="_example">
+        <title>Example</title>
+        <programlisting language="#{lang}" linenumbering="unnumbered"><ulink type="snippet" url="snippets/1.#{lang}"/>foo</programlisting>
+        </chapter>
+      DOCBOOK
+      actual = convert input, stub_file_opts, eq(
+        "INFO: <stdin>: line 3: writing snippet snippets/1.#{lang}"
+      )
+      expect(actual).to eq(expected.strip)
+    end
+  end
+
+  it "transforms SENSE comments into a listing with the SENSE language and a path" do
+  input = <<~ASCIIDOC
+    == Example
+    [source,js]
+    ----
+    foo
+    ----
+    // SENSE: snippet.sense
+  ASCIIDOC
+  expected = <<~DOCBOOK
+    <chapter id="_example">
+    <title>Example</title>
+    <programlisting language="sense" linenumbering="unnumbered"><ulink type="snippet" url="snippets/snippet.sense"/>foo</programlisting>
+    </chapter>
+  DOCBOOK
+  warnings = <<~WARNINGS
+    INFO: <stdin>: line 3: copying snippet #{spec_dir}/snippets/snippet.sense
+    WARN: <stdin>: line 3: reading snippets from a path makes the book harder to read
+  WARNINGS
+  actual = convert input, stub_file_opts, eq(warnings.strip)
+  expect(actual).to eq(expected.strip)
+end
```

Comment:
I see that your additions to this rspec file are consistent with the style of the rest of the tests but you can use `let` blocks and `shared_examples` to make it more readable and easier to extend in the future. I'm happy to discuss further in a call or in chat, as I don't think I can explain it very well in this comment.